### ISSUE TITLE
4.7, 4.8 Get Active Enrollments/Households - spec problem

### ISSUE THREAD

**Issue by michelleogburn** [abtassociates/LSASampleCode#34]
Apologies for the lateness... I've attempting to understand/verify the concern.

In the code for 4.7, there is the condition:
https://github.com/MollyMcEvilley/LSASampleCode/blob/b20ecb8fe54394498b739d006e2cf36a881e8c03/LSASampleCode.sql#L796
And in 4.7 the same condition:
https://github.com/MollyMcEvilley/LSASampleCode/blob/b20ecb8fe54394498b739d006e2cf36a881e8c03/LSASampleCode.sql#L834

However, in the spec, the restriction for [ExitDate] > [EntryDate] is missing:
![lsaspec4 7page23](https://user-images.githubusercontent.com/20728145/47468577-e15d3c80-d7b0-11e8-98ba-ff0b015ab969.png)

Without this condition, enrollments where [ExitDate] = [EntryDate] will not be excluded from the universe.

**Comment by MollyMcEvilley** [comment]
TL;DR version:  Changing criteria in the code so that for enrollments with non-null exit dates:

- _EntryDate_ must be >= ExitDate_ for strictly residential projects (ES/SH/TH) OR
- _ProjectType_ must be RRH/PSH. 

--------------------------

Regardless of a defined requirement in the specs, this is consistent with the HMIS Data Standards (see details below).  It is not necessary in systems that prevent users from entering entering an _ExitDate_ that is logically inconsistent with _EntryDate_ on the front end.  

The specs do note, in the context of a different instance in which HMIS data may be inconsistent with Dictionary requirements:

> In general, these specifications assume that HMIS data are consistent with Dictionary requirements. 

The HMIS Data Dictionary doesn't have a defined requirement for system logic that would prevent users from entering an _ExitDate_ that is logically inconsistent with _EntryDate_.  I can't remember if it ever came up in discussions of various drafts of the Dictionary, but I can say with certainty that the absence of a requirement to keep it from happening does not imply that it should be allowed to happen.  The Dictionary DOES note in Other System Issues that:

> If a client is in a project for a single day and has received some service but has not slept in a bed overnight (i.e., starts and stops before midnight of same day) the Project Exit Date may be the same as the Project Start Date. 

"Project Exit Date **may** be the same as Project Start Date" identifies an exception to the assumption that an exit takes place AFTER an entry.  
- Granted, this is not explicit.  I think the implication is clear, but I'm flagging this issue for review for LSA v2 and the next revision of the Dictionary.  
- I hope everyone would agree that **any** enrollment where _ExitDate_ is LESS THAN _EntryDate_ is... not good and should be excluded if it's existence is possible... but there's no requirement to prevent that, either.

- I can think of circumstances in which an RRH enrollment where ExitDate = EntryDate might possibly be legit.  I don't think it's logically possible that a PSH enrollment of this type could represent the provision of a real service provided to an eligible household, but the same period of services prior to move-in/possibility of exit without move-in does exist, so the code treats the two project types the same.

Writing the code was originally intended as an exercise to go through the business logic at the level of detail required by programming to identify factors that might not be apparent in thinking or talking about the logic abstractly _so I could make sure those factors/details were included in the specs_.  This is a perfect example of a very imperfect implementation of that intent -- i.e., I learned at some point of the possibility that records of this type might exist and accounted for it in the code but not the specs.  

Regardless of the original intent for the code, it was made public as a reference intended to illustrate the business logic of the specs and not to go rogue / sneak in undocumented requirements, and this isn't in the specs.  Still -- I can't see the utility of changing the code to include logically inconsistent data (if and where they exist).
